src/WebJobs.Script/Description/DotNet/CSharp/Analyzers/InvalidFileMetadataReferenceAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description.DotNet.CSharp.Analyzers
+{
+    [CLSCompliant(false)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class InvalidFileMetadataReferenceAnalyzer : DiagnosticAnalyzer
```

Comment:
Essentially, yes. You could still have analyzers that perform multiple related validations, but these are usually specialized types. The good thing is that we can actually package these analyzers and make them available so other tooling (e.g. VS) would be able to perform Azure Functions diagnostics as well.